src/bigwig/BigWigReader.ts

```diff
@@ -142,11 +149,13 @@ export class BigWigReader {
      * @param startBase Starting base pair
      * @param endChrom Ending chromose
      * @param endBase Ending base pair
+     * @param [restParser] Parser for reading data
      */
-    async readBigBedData(startChrom: string, startBase: number, endChrom: string, 
-            endBase: number): Promise<Array<BigBedData>> {
-        return this.readData<BigBedData>(startChrom, startBase, endChrom, endBase, 
-            (await this.getHeader()).common!.fullIndexOffset, decodeBedData);
+    async readBigBedData(startChrom: string, startBase: number, endChrom: string, endBase: number): Promise<Array<BigBedData>>;
+    async readBigBedData<T>(startChrom: string, startBase: number, endChrom: string, endBase: number, restParser: ParseFunction<T>): Promise<Array<T>>;
+    async readBigBedData<T>(startChrom: string, startBase: number, endChrom: string, endBase: number, restParser?: ParseFunction<T>): Promise<Array<(T | BigBedData)>> {
+        return this.readData(startChrom, startBase, endChrom, endBase,
+            (await this.getHeader()).common!.fullIndexOffset, { decodeFunction: decodeBedData, restParser: restParser || parseBigBed as any });
```

Comment:
This really doesn't end up being any different than a separate optional argument...you still end up having to pipe it through.

Reply:
Wow. I have never seen anyone use an arrow function to make a closure. Thanks for the insights and patience. They were very helpful. Hopefully now things are perfect.

Reply:
I personally prefer this syntax, but you could also do the same thing with:
```
function decodeBedData<T>(restParser: ParseFunction<T>) {
  return function(...): T[] {
  }
}
```